contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -0,0 +1,681 @@
+
+        // update cover amount
+        _updateActiveCoverLimit(0, coverLimit_);
+        _coverLimitOf[policyID] = coverLimit_;
+        _preDeactivateCoverLimitOf[policyID] = coverLimit_;
+        emit PolicyCreated(policyID);
+        return policyID;
+    }
+
+    /**
+     * @notice Updates the cover amount of your policy
+     * @notice If you update the cover limit for your policy, you will exit the cooldown process if you already started it. This means that if you want to withdraw all your funds, you have to redo the 'deactivatePolicy() => withdraw()' process
+     * @param newCoverLimit_ The new value to cover in **ETH**.
+     * @param referralCode_ Referral code
+    */
+    function updateCoverLimit(uint256 newCoverLimit_, uint256 referralCode_) external override nonReentrant whileUnpaused {
+        require(newCoverLimit_ > 0, "zero cover value");
+        uint256 policyID = _policyOf[msg.sender];
+        require(_exists(policyID), "invalid policy");
+        uint256 currentCoverLimit = coverLimitOf(policyID);
+        require(_canPurchaseNewCover(currentCoverLimit, newCoverLimit_), "insufficient capacity for new cover");
+        require(_accountBalanceOf[msg.sender] > _minRequiredAccountBalance(newCoverLimit_), "insufficient deposit for minimum required account balance");
+        
+        if (referralCode_ != 0) _processReferralCode(msg.sender, newCoverLimit_, referralCode_);
+        _exitCooldown(msg.sender); // Reset cooldown
+        _coverLimitOf[policyID] = newCoverLimit_;
+        _preDeactivateCoverLimitOf[policyID] = newCoverLimit_;
+        _updateActiveCoverLimit(currentCoverLimit, newCoverLimit_);
+        emit PolicyUpdated(policyID);
+    }
+
+    /**
+     * @notice Deposits funds for policy holders.
+     * @param policyholder_ The holder of the policy.
+    */
+    function deposit(address policyholder_) external payable override whileUnpaused {
+        _deposit(policyholder_, msg.value);
+    }
+
+    /**
+     * @notice Withdraw ETH from Soteria account to user.
+     * @notice If cooldown has passed, the user can withdraw any amount
+     * @notice If cooldown has not passed, the user can only withdraw down to minRequiredAccountBalance
+     * @param amount_ Amount policyholder desires to withdraw.
+     * User Soteria account must have > minAccountBalance.
+     * Otherwise account will be deactivated.
+     */
+    function withdraw(uint256 amount_) external override nonReentrant whileUnpaused {
+      require(amount_ <= _accountBalanceOf[msg.sender], "cannot withdraw > account balance");      
+      uint256 preDeactivateCoverLimit = _preDeactivateCoverLimitOf[_policyOf[msg.sender]];
+
+      if ( _hasCooldownPassed(msg.sender) ) {
+        _withdraw(msg.sender, amount_);
+      } else {
+        require(_accountBalanceOf[msg.sender] - amount_ > _minRequiredAccountBalance(preDeactivateCoverLimit), "must have > minRequiredAccountbalance");
+        _withdraw(msg.sender, amount_);
```

Comment:
We should default withdrawal to max available instead of reverting if amount > _minRequiredAccountBalance or amount > _accountBalanceOf

Reply:
Ok 👍